src/app/dynamic/enterprise/quotas/quota-widget/component.ts

```diff
@@ -0,0 +1,124 @@
+//                Kubermatic Enterprise Read-Only License
+//                       Version 1.0 ("KERO-1.0”)
+//                   Copyright © 2022 Kubermatic GmbH
+//
+// 1. You may only view, read and display for studying purposes the source
+//    code of the software licensed under this license, and, to the extent
+//    explicitly provided under this license, the binary code.
+// 2. Any use of the software which exceeds the foregoing right, including,
+//    without limitation, its execution, compilation, copying, modification
+//    and distribution, is expressly prohibited.
+// 3. THE SOFTWARE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND,
+//    EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+//    MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+//    IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
+//    CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+//    TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+//    SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+//
+// END OF TERMS AND CONDITIONS
+
+import {Component, Input, OnDestroy, OnInit, HostListener, ChangeDetectorRef} from '@angular/core';
+import {BehaviorSubject, Subject, lastValueFrom} from 'rxjs';
+import {debounceTime, take, takeUntil, map, filter} from 'rxjs/operators';
+import {QuotaDetails, QuotaVariables} from '@shared/entity/quota';
+import {isEnterpriseEdition} from '../../../common';
+import {getPercentage} from '@shared/utils/common';
+import {ThemePalette} from '@angular/material/core';
+import {QuotaService} from '../service';
+import {UserService} from '@core/services/user';
+import {Member} from '@shared/entity/member';
+
+@Component({
+  selector: 'km-quota-widget',
+  templateUrl: './template.html',
+  styleUrls: ['./style.scss'],
+})
+export class QuotaWidgetComponent implements OnInit, OnDestroy {
+  private readonly _unsubscribe = new Subject<void>();
+  private readonly _debounce = 500;
+  private readonly _showDetails$ = new BehaviorSubject(false);
+  private readonly _getIsEnterpriseEdition = isEnterpriseEdition;
+  private readonly _getPercentage = getPercentage;
+  private _user: Member;
+
+  @Input() projectId: string;
+  @Input() showCompactView = true;
+  @Input() showInCard = true;
+  @Input() showIcon = true;
+  @Input() showCompactViewDetailsOnHover = true;
```

Comment:
```suggestion
  @Input() showDetailsOnHover = true;
```

Reply:
`showDetailsOnHover` is not specific enough, as the `details view` won't show on hover if the widget is not in `compact view`